### PR TITLE
Rename (some) user-visible, non-code, trademarked names

### PR DESCRIPTION
**Pull request by lleyton**
As the title says. My poor computer is having trouble running all of the tests, so someone should probably check that before merge (CI is disabled atm.)

### CODE REVIEW

File: tests/rustdoc-ui/ice-bug-report-url.stderr [hunk 1]
```diff
@@ -11,7 +11,7 @@ error: the compiler unexpectedly panicked. this is a bug.
 
 note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-rustdoc&template=ice.md
 
-note: rustc {version} running on {platform}
+note: crabc 1.72.0-dev running on aarch64-apple-darwin
```
Comment (trvswgnr) [inline tests/rustdoc-ui/ice-bug-report-url.stderr:14]: do we want these to be explicitly set like this?

File: tests/ui/const-generics/late-bound-vars/in_closure.stderr [hunk 2]
```diff
@@ -1,4 +1,6 @@
-error: query stack during panic:
+error: note: crabc 1.72.0-dev running on aarch64-apple-darwin
```
Comment (trvswgnr) [inline tests/ui/const-generics/late-bound-vars/in_closure.stderr:1]: same thing here, were these autogenerated from running a test or somethin?

File: tests/ui/chalkify/bugs/async.stderr [hunk 3]
```diff
@@ -11,7 +11,9 @@ error: internal compiler error: projection clauses should be implied from elsewh
   --> $DIR/async.rs:23:25
    |
 LL | async fn foo(x: u32) -> u32 {
-   |                         ^^^query stack during panic:
+   |                         ^^^note: crabc 1.72.0-dev running on aarch64-apple-darwin
```
Comment (trvswgnr) [inline tests/ui/chalkify/bugs/async.stderr:14]: +++

File: tests/ui/chalkify/bugs/async.stderr [hunk 4]
```diff
@@ -11,7 +11,9 @@ error: internal compiler error: projection clauses should be implied from elsewh
...
+
+query stack during panic:
```
Comment (trvswgnr) [inline tests/ui/chalkify/bugs/async.stderr:16]: +++

**Review by trvswgnr — CHANGES_REQUESTED** [review]
looks like some of these files have auto-generated changes that should be placeholders.

overall it looks good though, and it compiles and runs with updated help docs, etc...

i ran some some of the `compiletest` suite and some failed. they passed when i passed the `--bless` flag.

**Comment by lleyton** [comment]
Yeah, all of those stderr's were generated when running the tests with `--bless`.

**Comment by noctisatrae** [comment]
So we did agree on a name for Crab's tools?

**Comment by trvswgnr** [comment]
> So we did agree on a name for Crab's tools?

there was a lot of discussion initially about what the tools should be named, and we came to a general consensus. i'm open to suggestions and/or renaming in the future but in the interest of keeping things moving we'll stick with these for now.

- rust    -> crab
- rustc   -> crabc
- rustfmt -> crabfmt
- rustup  -> crabup
- cargo   -> crabgo
- clippy  -> pinchy
- crate   -> claw

if you want to discuss it further we can use GH Discussions or make a post in discord

**Review by trvswgnr — APPROVED** [review]
looks good now!

**Pull request merged**